gst/nnstreamer/tensor_converter/tensor_converter.c

```diff
@@ -838,46 +838,50 @@ _gst_tensor_converter_chain_timestamp (GstTensorConverter * self,
   self->old_timestamp = GST_BUFFER_TIMESTAMP (inbuf);
 }
 
-/** @brief Chain function's private routine to push buffer into src pad */
-static GstFlowReturn
-_gst_tensor_converter_chain_push (GstTensorConverter * self, GstBuffer * buf)
+/** @brief Chain function's private routine to process multi tensor */
+static GstBuffer *
+_gst_tensor_converter_chain_multi_tensor (GstTensorConverter * self,
+    GstBuffer * buf)
 {
   GstBuffer *buffer = buf;
 
-  if (self->in_media_type == _NNS_OCTET) {
-    /* configure multi tensors */
-    if (self->tensors_info.num_tensors > 1) {
-      GstMemory *mem;
-      GstMapInfo info;
-      gsize idx, size;
-      guint i;
+  /* configure multi tensors */
+  if (self->tensors_info.num_tensors > 1) {
+    GstMemory *mem;
+    gsize idx, size;
+    guint i;
 
-      g_assert (self->frames_per_tensor == 1);
+    g_assert (self->frames_per_tensor == 1);
 
-      mem = gst_buffer_get_all_memory (buf);
-      if (!gst_memory_map (mem, &info, GST_MAP_READ)) {
-        ml_logf ("Failed to map the input buffer for octet stream.");
-        gst_memory_unref (mem);
-        return GST_FLOW_ERROR;
-      }
+    idx = 0;
+    buffer = gst_buffer_new ();
+    mem = gst_buffer_get_all_memory (buf);
 
-      idx = 0;
-      buffer = gst_buffer_new ();
+    for (i = 0; i < self->tensors_info.num_tensors; ++i) {
+      size = gst_tensors_info_get_size (&self->tensors_info, i);
+      gst_buffer_append_memory (buffer, gst_memory_share (mem, idx, size));
+      idx += size;
```

Comment:
It's not related to this PR but I think `idx` is confusing. Better to use `offset`?

Reply:
updated